src/ga/ga_selectors.rs

```diff
@@ -136,55 +128,46 @@ impl<'a, T: GASolution> GASelector<'a, T> for GARankSelector<'a, T>
         }
 
         // Select any individual from those that share best score.
-        self.population.individual(ga_random::ga_random_range(0, best_count), population_sort_basis)
+        population.individual(ga_random::ga_random_range(0, best_count), population_sort_basis)
     }
 }
 
-pub struct GAUniformSelector<'a, T: 'a + GASolution>
+pub struct GAUniformSelector
 {
-    population: &'a mut GAPopulation<T>
 }
 
-impl<'a, T: GASolution> GAUniformSelector<'a, T>
+impl GAUniformSelector
 {
-    pub fn new(p: &'a mut GAPopulation<T>) -> GAUniformSelector<'a, T>
+    pub fn new() -> GAUniformSelector
     {
         GAUniformSelector
         {
-            population: p
         }
     }
 }
 
-impl<'a, T: GASolution> GASelector<'a, T> for GAUniformSelector<'a, T>
+impl<'a, T: GASolution> GASelector<'a, T> for GAUniformSelector
 {
-    fn assign(&mut self, population: &'a mut GAPopulation<T>)
-    {
-        self.population = population;
-    }
-
-    fn update(&mut self)
+    fn update(&mut self, population : &mut GAPopulation<T>)
     {
         // Need to sort first, because GAPopulation.individual() draws individuals
         // from the sorted lists.
-        self.population.sort();
+        population.sort();
```

Comment:
@Sysnett Now that population is not a member of a selector, it seems a little strange to have the selector sort the population. It made sense when the population was part of the selector's state. There is no guarantee anymore that the "updated" population will be the one used by a subsequent select() call.

It wouldn't be right, either, to sort it inside select(), as there may be multiple select() calls on a population whose order doesn't change between such calls.

The other option is to have client code call sort() before a series of select() calls, but how to enforce that? Need to think.